redpanda/redpanda.go

```diff
@@ -117,11 +131,22 @@ type credentials struct {
 func getCredentials(ctx context.Context, cloudEnv string, conf models.Redpanda) (creds credentials, diags diag.Diagnostics) {
 	// Similar to other Terraform providers, this provider (1) can pick up authentication configuration
 	// from multiple sources, and (2) gives precedence to direct configuration over environment variables.
-	// The first source which provides a client_id or client_secret will be chosen to validate and
+	// The first source which provides a client_id, client_secret, or token will be chosen to validate and
 	// try to use. The sources are checked in the following order:
 	// 1. Parameters in the provider configuration
 	// 2. Environment variables
 
+	// TODO: There are some scenarios here that could be handled more smoothly.
+	// - If a source has both a token and a client id/secret pair, should we validate the token 'sub'/'azp'
+	//   claim against the client id?
+	// - If the token is expired, should we try to renew it using a client id/secret pair? Or bail?
+	// - If the provider configuration has a client id/secret pair, but a token is provided in an environment
+	//   variable, should we use the token if it validates against the client id?
+	// - Should we keep the client id/secret around as well in case we need to re-auth with the API?
+	// - If the user passes in a token, should we validate its audience field against the configured
+	//   Redpanda cloud environment/endpoint? or, should we pull the endpoint from the audience field? or
+	//   do nothing like we do now?
+
 	endpoint, err := cloud.EndpointForEnv(cloudEnv)
 	if err != nil {
 		diags.AddError(err.Error(), "")
```

Comment:
We really should have a searchable error summary message here

Reply:
For the token vs. environment bit specifically?